pkg/actuators/machine/vm.go

```diff
@@ -186,6 +210,58 @@ func validateSubnetConfig(mscp *machineScope, subnet *machinev1.NutanixResourceI
 	return nil
 }
 
+func validateProjectConfig(mscp *machineScope, fldPath *field.Path) *field.Error {
+	var err error
+	var errMsg string
+
+	switch mscp.providerSpec.Project.Type {
+	case "":
+		// ignore if not configured
+		return nil
+	case machinev1.NutanixIdentifierName:
+		if mscp.providerSpec.Project.Name == nil || *mscp.providerSpec.Project.Name == "" {
+			return field.Required(fldPath.Child("project", "name"), "Missing projct name")
+		} else {
+			projectName := *mscp.providerSpec.Project.Name
+			projectRefUuidPtr, err := findProjectUuidByName(mscp.nutanixClient, projectName)
+			if err != nil {
+				errMsg = fmt.Sprintf("Failed to find project with name %q. error: %v", projectName, err)
+				return field.Invalid(fldPath.Child("project", "name"), projectName, errMsg)
+			} else {
+				mscp.providerSpec.Project.Type = machinev1.NutanixIdentifierUUID
+				mscp.providerSpec.Project.UUID = projectRefUuidPtr
```

Comment:
Why are we modifying the providerSpec? I would expect if the user has set the name that we look up the UUID every time, rather than modifying their spec for them

Reply:
The Machine CR's providerSpec is not modified anywhere in the mapi-nutanix-controller, it is just used by reading/checking purpose. And since when creating the VM, the prism-api requires the VM resources' UUID to be set (name is not required, just used to look up the resource UUID), we just set the UUID type at validation time, to avoid calling the prism-api to look up the resource UUID again later.

Reply:
I would suggest it's probably best to write something dedicated into the machine scope for this rather than modifying the `machineScope.providerSpec`, my concern is that, while we may not be doing it now, maybe in the future we start writing the providerSpec back to the Machine, at which point we will likely break things by modifying the spec in this code

Reply:
Okay, will fix this.

Reply:
i tend to agree with Joel here, but i want to make sure i understand how the UUID value is being used. @yanhua121 is the `mscp.providerSpec.Project.UUID` being written back to kubernetes or is just being used as a temporary value in the controller here?

Reply:
@JoelSpeed Fixed as you suggested.
@elmiko Currently, the providerSpec in the MachineScope struct is not written back to the Machine CR object. But as Joel pointed out, we need to leave room for the future to write it back.